### PR TITLE
Added subsampling support to max_pool_2d

### PR DESCRIPTION
**Pull request by sergsb**
There is a new option in theano 0.7 in max_pool_2d - st, so we able to make stride.

### CODE REVIEW

File: keras/layers/convolutional.py [hunk 1]
```diff
@@ -135,20 +135,21 @@ def get_config(self):
 
 
 class MaxPooling2D(Layer):
-    def __init__(self, poolsize=(2, 2), ignore_border=True):
+    def __init__(self, poolsize=(2, 2), ignore_border=True,subsample=None):
```
Comment (fchollet) [inline keras/layers/convolutional.py]: Please add spaces after commas.

**Comment by fchollet** [comment]
For the record, here's the documentation: http://deeplearning.net/software/theano/library/tensor/signal/downsample.html

"st (tuple of lenght 2) – stride size, which is the number of shifts over rows/cols to get the the next pool region. if st is None, it is considered equal to ds (no overlap on pooling regions)"

We can definitely add this to MaxPooling2D, as long it is properly explained in the documentation.

**Comment by sergsb** [comment]
Fixed

File: keras/layers/convolutional.py (hunk 1 above)
Comment (sergsb) [inline keras/layers/convolutional.py]: Fixed

**Comment by fchollet** [comment]
I think the `st` parameter was a poor naming choice of the part of the Theano devs; it is absolutely not obvious at first what it stands for (stride) or what it does. 

We should definitely rename it, and write a proper documentation paragraph for the MaxPooling2D layer.

**Pull request closed without merging**